src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -882,22 +879,19 @@ def generic_wait_command(self, name, getter_name='get', getter_type=None, **kwar
 
         if getter_type:
             merged_kwargs = _merge_kwargs(getter_type.settings, merged_kwargs, CLI_COMMAND_KWARGS)
-        getter_op = self._resolve_operation(merged_kwargs, getter_name, getter_type)
-        _cli_generic_wait_command(
-            self.command_loader,
-            '{} {}'.format(self.group_name, name),
-            getter_op=getter_op,
-            **merged_kwargs)
+        getter_op = self._resolve_operation(merged_kwargs, getter_name, getter_type, custom_type=custom_command)
+        _cli_generic_wait_command(self.command_loader, '{} {}'.format(self.group_name, name), getter_op=getter_op,
+                                  custom_command=custom_command, **merged_kwargs)
 
-    def generic_show_command(self, name, getter_name='get', getter_type=None, **kwargs):
+    def generic_show_command(self, name, getter_name='get', getter_type=None, custom_command=False, **kwargs):
```

Comment:
Is there anything actually generic about this?  In the case of update, the generic piece are the `--set, --add, --remove` arguments. In the case of wait, it's the different generic wait conditions.

My preference here would be to expose `show_command` and `custom_show_command` which eliminates the need for the `custom_command` kwarg and neatly parallels the `command`/`custom_command` pattern. 

I would leave generic update/wait as is.

Reply:
Hmm, depends on how we define "generic". It's generic in terms of providing a base functionality for all commands registered with it. The signature for the commands are also similar.
It does not, however, provide any additional arguments.

I like the idea of exposing `show_command`/`custom_show_command` but it does seem inconsistent if update/wait do not follow the `*_command`/`custom_*_command` pattern.

@troydai @yugangw-msft, what do you guys think about the addition of a 'custom_command' kwarg?

Currently, `generic_*` commands will only make use of the `command_type` kwarg. `custom_command_type`, whether defined in the command_loader, or the command_group() is ignored. The `custom_command` kwarg will refer to the `custom_command_type` by default, for cases like [this](https://github.com/Azure/azure-cli/blob/dev/src/command_modules/azure-cli-eventgrid/azure/cli/command_modules/eventgrid/commands.py#L35-L45).

Reply:
They are dubbed generic because of the syntax (especially for "generic" update), not because they enforce some common behavior. The add arguments that may not actually work for a given situation (for example, you create a resource with --no-wait and then wait on --deleted... dumb thing to do, but the command can't know).

Reply:
I do agree that `wait` and `show` should use the same mechanisms, because in probably 99% of cases they likely are using the exact same getters. I would rather advocate for this scheme in lieu of the flag:
```
show_command(…)
custom_show_command(…)
wait_command(…)
custom_wait_command(…)
generic_wait_command(…) # deprecate and phase out
generic_update_command(…) # remains unaltered
```

The main difference is that show and wait only rely on a single callable, so it is appropriate to say whether that kwarg bundle comes from the normal `command_type` bundle or the `custom_command_type` bundle since we do that for 'regular' commands. 

However, generic update weaves together 2-3 independent callables to accomplish something. There should only be one default for this, and that default, which fits 90% of the cases is that the setter and getter come from `command_type` and the custom piece comes from `custom_command_type`. If you deviate from that pattern, you have to get specific. I'd bet at least half of the usages of generic update that use custom getters and setters could be rewritten to not use them at all. There are a small number of cases where Swagger naming issues necessitate using the name kwarg, but the default still holds for the types.